mreg_cli/utilities/api.py

```diff
@@ -80,18 +90,29 @@ def login1(user: str, url: str) -> None:
         error(f"Could not connect to {url}")
 
     if ret.status_code == 401:
-        login(user, url)
+        prompt_for_password_and_login(user, url, catch_exception=False)
+
 
+def prompt_for_password_and_login(user: str, url: str, catch_exception: bool = True) -> None:
+    """Login to MREG.
 
-def login(user: str, url: str) -> None:
-    """Login to MREG."""
+    :param user: Username to login with.
+    :param url: URL to MREG.
+    :param catch_exception: If True, login errors are caught, otherwise they are passed on.
+
+    :raises LoginFailedError: If login fails and catch_exception is False.
+
+    :returns: Nothing.
+    """
     print(f"Connecting to {url}")
-    # get url
     password = prompt(f"Password for {user}: ", is_password=True)
     try:
-        _update_token(user, password)
+        auth_and_update_token(user, password)
     except CliError as e:
-        e.print_self()
+        if catch_exception:
+            e.print_self()
```

Comment:
Note: would be nice if we had unit tests that ensured we print something useful here.

I think it's fine for now, but this behavior is a bit opaque from the caller's perspective.

Reply:
Well, mreg-cli should have unit tests. :)